### PR TITLE
[REVIEW] Update conda dev yml files

### LINKED ISSUE #301: [BUG] No yml file for CUDA 11.0 dev environment
The yml file for creating a dev environment with CUDA 11.0 is not present in the `conda/environments/`  folder.

### PR DESCRIPTION
**Pull request by Salonijain27**
Closes #301 
removed files used to build a conda dev environment for CUDA 9.2 and CUDA 10.0 
added a file for CUDA 11.0
added cmake and cython libraries into the yml files
updated cudf to 0.16.* from 0.15

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by harrism — APPROVED** [review]
Looks good. Thanks @Salonijain27 !

**Review by ajschmidt8 — APPROVED** [review]

File: conda/environments/cuspatial_dev_cuda10.1.yml [hunk 1]
```diff
@@ -8,7 +8,10 @@ channels:
 dependencies:
   - clang=8.0.1
   - clang-tools=8.0.1
-  - cudf=0.15.*
+  - cudf=0.16.*
   - cudatoolkit=10.1
   - gdal>=3.0.2
   - geopandas=0.7.0
+  - cmake>=3.14
+  - cmake_setuptools>=0.1.3
```
Comment (harrism) [inline conda/environments/cuspatial_dev_cuda10.1.yml:16]: @kkraus14 says we don't need this.
```suggestion
```

File: conda/environments/cuspatial_dev_cuda10.2.yml [hunk 2]
```diff
@@ -8,7 +8,10 @@ channels:
 dependencies:
   - clang=8.0.1
   - clang-tools=8.0.1
-  - cudf=0.15.*
+  - cudf=0.16.*
   - cudatoolkit=10.2
   - gdal>=3.0.2
   - geopandas=0.7.0
+  - cmake>=3.14
+  - cmake_setuptools>=0.1.3
```
Comment (harrism) [inline conda/environments/cuspatial_dev_cuda10.2.yml:16]:
```suggestion
```

File: conda/environments/cuspatial_dev_cuda11.0.yml [hunk 3]
```diff
@@ -8,7 +8,10 @@ channels:
...
   - clang=8.0.1
   - clang-tools=8.0.1
-  - cudf=0.15.*
-  - cudatoolkit=10.0
+  - cudf=0.16.*
+  - cudatoolkit=11.0
   - gdal>=3.0.2
   - geopandas=0.7.0
+  - cmake>=3.14
+  - cmake_setuptools>=0.1.3
```
Comment (harrism) [inline conda/environments/cuspatial_dev_cuda11.0.yml:16]:
```suggestion
```

**Review by harrism — CHANGES_REQUESTED** [review]

**Review by kkraus14 — APPROVED** [review]

**Comment by harrism** [comment]
rerun tests

**Review by kkraus14 — APPROVED** [review]

**Pull request merged**